integrations/Matomo/util.js

```diff
@@ -0,0 +1,316 @@
+      if (!orderId) {
+        logger.error("User parameter order_id is required");
+        break;
+      }
+
+      grandTotal =
+        get(message, "properties.total") || get(message, "properties.revenue");
+      /** grandTotal is a mandatory field
+       * If grandTotal is not found inside the properties, we are calculating it manually
+       */
+      if (!grandTotal) {
+        if (products && Array.isArray(products) && products.length > 0) {
+          grandTotal = 0;
+          // iterating through the products array and calculating grandTotal
+          for (let product = 0; product < products.length; product++) {
+            if (product.price && product.quantity) {
+              if (typeof price === "string") price = parseFloat(price);
+              if (typeof quantity === "string") quantity = parseFloat(quantity);
+              grandTotal += product.price * product.quantity;
+            }
+          }
+        }
+      }
+
+      /** subTotal is not a listed property in "Order Completed" event
+       * if user doesn't provide this property, then we are calculating it from grandTotal
+       * ref: https://matomo.org/faq/reports/analyse-ecommerce-reporting-to-improve-your-sales/#conversions-overview
+       */
+      subTotal = get(message, "properties.subTotal");
+      if (!subTotal) {
+        subTotal = grandTotal;
+      }
+      tax = get(message, "properties.tax");
+      shipping = get(message, "properties.shipping");
+      discount = get(message, "properties.discount");
+
+      // ref: https://matomo.org/faq/reports/analyse-ecommerce-reporting-to-improve-your-sales/#conversions-overview
+      if (shipping) {
+        if (typeof shipping === "string") shipping = parseFloat(shipping);
+        grandTotal += shipping;
+      }
+      if (tax) {
+        if (typeof tax === "string") tax = parseFloat(tax);
+        grandTotal += tax;
+      }
+
+      window._paq.push([
+        "trackEcommerceOrder",
+        orderId,
+        grandTotal,
+        subTotal,
+        tax,
+        shipping,
+        discount,
+      ]);
+      window._paq.push(["clearEcommerceCart"]);
+
+      break;
+
+    default:
+      // Generic track Event
+      category = get(message, "properties.category");
+      if (!category) {
+        logger.error("User parameter category is required");
+        break;
+      }
+
+      action = get(message, "properties.action");
+      if (!action) {
+        logger.error("User parameter action is required");
+        break;
+      }
+
+      name = get(message, "event");
+      value = get(message, "properties.value");
+      window._paq.push(["trackEvent", category, action, name, value]);
+      break;
+  }
+};
+
+/**
+ * Checks for custom dimensions in the payload
+ * If present, we make setCustomDimension() function call
+ * @param  {} message
+ */
+const checkCustomDimensions = (message) => {
+  const customDimensions = get(message, "integrations.MATOMO.customDimension");
```

Comment:
This is not the correct way to get integrations object. Lets write a small util function like transformer to use commin names to get this
https://github.com/rudderlabs/rudder-transformer/blob/febc3b89209bf64f086378aab77d7c743c81dcd4/v0/util/index.js#L831

Reply:
Done